integration-tests/tests/mysql.test.ts

```diff
@@ -1967,3 +1967,46 @@ test.serial('update undefined', async (t) => {
 
 	await db.execute(sql`drop table ${users}`);
 });
+
+test.serial('utc config for datetime', async (t) => {
```

Comment:
Could you add a check to this test that verifies the actual value that's put into the DB with `mode: "date"`? You could use a raw `select` query to get the data bypassing the mapping.